pkg/certificate/authorization/service.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package authorization
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"storj.io/storj/pkg/pb"
+)
+
+// ErrService is the default error class for the authorization service.
+var ErrService = errs.Class("authorization service error")
+
+// Service is the authorization service.
+type Service struct {
+	log *zap.Logger
+	db  *DB
+}
+
+// NewService creates a new authorization service.
+func NewService(log *zap.Logger, db *DB) *Service {
```

Comment:
The endpoint doesn't need to know about that logic but it makes sense that it handles the protobuf request and response types. I'll make this nicer.

Reply:
I thought that they don't have to return gRPC status codes because they shouldn't depend on the transport/communication protocol. However, in my opinion, that line is violated accepting protocol buffers structures, but Storj services use them.